### PR TITLE
ENH: validate the zoom level

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
xref https://github.com/darribas/contextily/issues/67

(doesn't fully close the issue, as I don't clip the zoom level if it is above automatically)

### CODE REVIEW

File: contextily/plotting.py [hunk 1]
```diff
@@ -110,11 +117,7 @@ def add_basemap(
             left, right, bottom, top = _reproj_bb(
                 left, right, bottom, top, crs, {"init": "epsg:3857"}
             )
-        # Zoom
-        if isinstance(zoom, str) and (zoom.lower() == "auto"):
-            min_ll = _sm2ll(left, bottom)
-            max_ll = _sm2ll(right, top)
-            zoom = _calculate_zoom(*min_ll, *max_ll)
```
Comment (jorisvandenbossche) [inline contextily/plotting.py]: this is already done in `bounds2img`

File: contextily/tile.py [hunk 2]
```diff
@@ -90,8 +90,6 @@ def bounds2raster(
         # Convert w, s, e, n into lon/lat
         w, s = _sm2ll(w, s)
         e, n = _sm2ll(e, n)
-    if zoom == "auto":
-        zoom = _calculate_zoom(w, s, e, n)
```
Comment (jorisvandenbossche) [inline contextily/tile.py]: again, this is already done in `bounds2img`

**Comment by darribas** [comment]
This looks great @jorisvandenbossche, thanks very much!!! I think this in itself is useful functionality so would be happy to merge, unless you want to keep pushing and add a re-assignment of `zoom` into the closest possible level (with a warning that that's happened)? Happy either way.

**Comment by jorisvandenbossche** [comment]
@darribas I added the logic to re-assign zoom to the max level, in case a too large value was inferred and a max_zoom is known.

**Pull request merged**